File: C++/Easy/WrongAnswer/circular-sentence_1.cpp
class Solution {
public:
    bool isCircularSentence(string sentence) {
        
        int n = sentence.size();
        int cnt = 0;
        for(int i = 0; i < n; i++){
            if(sentence[i] == ' '){
                cnt++;
            }
        }
        if(cnt == 0) return true;
        int i = 0, j = n - 1;
        while(i < n && sentence[i] == ' ') i++;
        while(j >= 0 && sentence[j] == ' ') j--;
        if(sentence[i] == sentence[j]) return true;
        return false;
    }
};

File: Rust/Easy/Accepted/circular-sentence_1.rs
impl Solution {
    pub fn is_circular_sentence(sentence: String) -> bool {
        
        let mut words = sentence.split(" ");
        let first = words.next().unwrap();
        let mut last = first;
        for word in words {
            if word.chars().next().unwrap() != last.chars().last().unwrap() {
                return false;
            }
            last = word;
        }
        first.chars().next().unwrap() == last.chars().last().unwrap()
    }
}